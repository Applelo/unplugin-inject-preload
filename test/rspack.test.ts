import { join } from 'node:path'
import { readFileSync } from 'node:fs'
import { describe, expect, it } from 'vitest'
import type { Configuration } from '@rspack/core'
import { rspack } from '@rspack/core'
import HtmlWebpackPlugin from 'html-webpack-plugin'
import UnpluginInjectPreload from './../src/rspack'
import type { Options } from './../src/types'
import configs from './fixtures/configs'

async function buildRspack(pluginConfig: Options, config: Configuration = {}) {
  return new Promise((resolve, reject) => {
    const compiler = rspack({
      mode: 'production',
      context: join(__dirname, 'fixtures/webpack'),
      entry: join(__dirname, 'fixtures/src/main.ts'),
      output: {
        path: join(__dirname, 'fixtures/webpack/dist'),
        publicPath: config.output?.publicPath || '/',
        clean: true,
        assetModuleFilename: '[name]-[hash][ext][query]',
      },
      resolve: {
        extensions: ['.ts', '.js'],
        extensionAlias: {
          '.js': ['.js', '.ts'],
        },
      },
      module: {
        rules: [
          {
            test: /\.css$/i,
            type: 'css',
          },
          {
            test: /\.(woff|woff2|eot|ttf|otf)$/i,
            type: 'asset/resource',
          },
        ],
      },
      plugins: [
        // @ts-expect-error work as expected
        new HtmlWebpackPlugin({
          minify: false,
          inject: false,
          template: join(__dirname, 'fixtures/webpack/index.html'),
        }),
        UnpluginInjectPreload(pluginConfig),
      ],
    })

    compiler.run((err, stats) => {
      if (err)
        reject(err)

      const statsErrors = stats ? stats?.toJson().errors : []
      if (statsErrors && statsErrors.length > 0) {
        console.error(statsErrors)
        reject(statsErrors)
      }

      const result = readFileSync(
        join(__dirname, 'fixtures/webpack/dist/index.html'),
        'utf8',
      )

      resolve(result)
    })
  })
}

describe('excerpt rspack', () => {
  for (const key in configs) {
    if (Object.prototype.hasOwnProperty.call(configs, key)) {
      const config = configs[key]
      it(`test ${key}`, async () => {
        const output = await buildRspack(config)
        expect(output).toMatchSnapshot()
      }, 8000)

      it(`test ${key} with basePath`, async () => {
        const output = await buildRspack(config, { output: { publicPath: '/base' } })
        expect(output).toMatchSnapshot()
      }, 8000)
    }
  }
})
