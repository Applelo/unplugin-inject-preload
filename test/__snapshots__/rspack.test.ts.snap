// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expect rspack with HtmlRspackPlugin > test auto 1`] = `
"<!DOCTYPE html>
<html>
<head>
  <link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font">
  <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font">

    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/main.js" defer></script><link href="/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test auto with basePath 1`] = `
"<!DOCTYPE html>
<html>
<head>
  <link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font">
  <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font">

    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/base/main.js" defer></script><link href="/base/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test customAttributes 1`] = `
"<!DOCTYPE html>
<html>
<head>
  <link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto">
  <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto">

    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/main.js" defer></script><link href="/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test customAttributes with basePath 1`] = `
"<!DOCTYPE html>
<html>
<head>
  <link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto">
  <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto">

    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/base/main.js" defer></script><link href="/base/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test customPosition 1`] = `
"<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font" data-vite-plugin-inject-preload>
    <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font" data-vite-plugin-inject-preload>

  <title>Unplugin Inject Preload</title><script src="/main.js" defer></script><link href="/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test customPosition with basePath 1`] = `
"<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font" data-vite-plugin-inject-preload>
    <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font" data-vite-plugin-inject-preload>

  <title>Unplugin Inject Preload</title><script src="/base/main.js" defer></script><link href="/base/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test entryMatch 1`] = `
"<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/main.js" defer></script><link href="/main.css" rel="stylesheet" />  <link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font">
  <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font">
</head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test entryMatch with basePath 1`] = `
"<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/base/main.js" defer></script><link href="/base/main.css" rel="stylesheet" />  <link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font">
  <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font">
</head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test injectBottom 1`] = `
"<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/main.js" defer></script><link href="/main.css" rel="stylesheet" />  <link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font">
  <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font">
</head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test injectBottom with basePath 1`] = `
"<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/base/main.js" defer></script><link href="/base/main.css" rel="stylesheet" />  <link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font">
  <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font">
</head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test modulepreload 1`] = `
"<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/main.js" defer></script><link href="/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test modulepreload with basePath 1`] = `
"<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/base/main.js" defer></script><link href="/base/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test noMatch 1`] = `
"<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/main.js" defer></script><link href="/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test noMatch with basePath 1`] = `
"<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/base/main.js" defer></script><link href="/base/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test noType 1`] = `
"<!DOCTYPE html>
<html>
<head>
  <link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" as="font">
  <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" as="font">

    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/main.js" defer></script><link href="/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test noType with basePath 1`] = `
"<!DOCTYPE html>
<html>
<head>
  <link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" as="font">
  <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" as="font">

    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/base/main.js" defer></script><link href="/base/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test wrongAttributes 1`] = `
"<!DOCTYPE html>
<html>
<head>
  <link rel="preload" href="./yolo.woff2" type="font/woff2" as="font">
  <link rel="preload" href="./yolo.woff2" type="font/woff2" as="font">

    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/main.js" defer></script><link href="/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlRspackPlugin > test wrongAttributes with basePath 1`] = `
"<!DOCTYPE html>
<html>
<head>
  <link rel="preload" href="./yolo.woff2" type="font/woff2" as="font">
  <link rel="preload" href="./yolo.woff2" type="font/woff2" as="font">

    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--__unplugin-inject-preload__-->
  <title>Unplugin Inject Preload</title><script src="/base/main.js" defer></script><link href="/base/main.css" rel="stylesheet" /></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
  

</body></html>"
`;

exports[`expect rspack with HtmlWebpackPlugin > test auto 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font"><link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><script defer src="/main.js"></script><link href="/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font"><link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><script defer src="/main.js"></script><link href="/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test auto with basePath 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font"><link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font"><link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test customAttributes 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto"><link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto"><script defer src="/main.js"></script><link href="/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto"><link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto"><script defer src="/main.js"></script><link href="/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test customAttributes with basePath 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto"><link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto"><script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto"><link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font" crossorigin="anonymous" data-font="Roboto"><script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test customPosition 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <script defer src="/main.js"></script><link href="/main.css" rel="stylesheet">

    <link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font" data-vite-plugin-inject-preload>
    <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font" data-vite-plugin-inject-preload>

  <script defer src="/main.js"></script><link href="/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test customPosition with basePath 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet">

    <link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font" data-vite-plugin-inject-preload>
    <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font" data-vite-plugin-inject-preload>

  <script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test entryMatch 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <script defer src="/main.js"></script><link href="/main.css" rel="stylesheet"><link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font">
    <!--__unplugin-inject-preload__-->
  <script defer src="/main.js"></script><link href="/main.css" rel="stylesheet"><link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test entryMatch with basePath 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet"><link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font">
    <!--__unplugin-inject-preload__-->
  <script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet"><link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test injectBottom 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <script defer src="/main.js"></script><link href="/main.css" rel="stylesheet"><link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font">
    <!--__unplugin-inject-preload__-->
  <script defer src="/main.js"></script><link href="/main.css" rel="stylesheet"><link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test injectBottom with basePath 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet"><link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font">
    <!--__unplugin-inject-preload__-->
  <script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet"><link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" type="font/woff2" as="font"><link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" type="font/woff2" as="font"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test modulepreload 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <script defer src="/main.js"></script><link href="/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <script defer src="/main.js"></script><link href="/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test modulepreload with basePath 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test noMatch 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <script defer src="/main.js"></script><link href="/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <script defer src="/main.js"></script><link href="/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test noMatch with basePath 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test noType 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" as="font"><link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" as="font"><script defer src="/main.js"></script><link href="/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <link rel="preload" href="/Roboto-Regular-ec5068bd8823a38af494.woff2" as="font"><link rel="preload" href="/Roboto-Italic-96e486355f319e0b182b.woff2" as="font"><script defer src="/main.js"></script><link href="/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test noType with basePath 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" as="font"><link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" as="font"><script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <link rel="preload" href="/base/Roboto-Regular-ec5068bd8823a38af494.woff2" as="font"><link rel="preload" href="/base/Roboto-Italic-96e486355f319e0b182b.woff2" as="font"><script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test wrongAttributes 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <link rel="preload" href="./yolo.woff2" type="font/woff2" as="font"><link rel="preload" href="./yolo.woff2" type="font/woff2" as="font"><script defer src="/main.js"></script><link href="/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <link rel="preload" href="./yolo.woff2" type="font/woff2" as="font"><link rel="preload" href="./yolo.woff2" type="font/woff2" as="font"><script defer src="/main.js"></script><link href="/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;

exports[`expect rspack with HtmlWebpackPlugin > test wrongAttributes with basePath 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Unplugin Inject Preload</title>
    <link rel="preload" href="./yolo.woff2" type="font/woff2" as="font"><link rel="preload" href="./yolo.woff2" type="font/woff2" as="font"><script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet">
    <!--__unplugin-inject-preload__-->
  <link rel="preload" href="./yolo.woff2" type="font/woff2" as="font"><link rel="preload" href="./yolo.woff2" type="font/woff2" as="font"><script defer src="/base/main.js"></script><link href="/base/main.css" rel="stylesheet"></head>
  <body>
    <h1>Unplugin Inject Preload</h1>
    
  </body>
</html>
"
`;
